contracts/services/astroport-lper/src/contract.rs

```diff
@@ -0,0 +1,379 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_json_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+
+use crate::{
+    error::{ServiceError, UnauthorizedReason},
+    msg::{Config, ExecuteMsg, InstantiateMsg, QueryMsg, ServiceConfigValidation},
+    state::{CONFIG, PROCESSOR},
+};
+
+// version info for migration info
+const CONTRACT_NAME: &str = env!("CARGO_PKG_NAME");
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ServiceError> {
+    cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    cw_ownable::initialize_owner(deps.storage, deps.api, Some(&msg.owner))?;
+
+    PROCESSOR.save(deps.storage, &deps.api.addr_validate(&msg.processor)?)?;
+
+    let config = msg.config.validate(deps.as_ref())?;
+    CONFIG.save(deps.storage, &config)?;
+
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("owner", msg.owner))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ServiceError> {
+    match msg {
+        ExecuteMsg::ProcessAction(action_msgs) => {
+            let processor = PROCESSOR.load(deps.storage)?;
+            if info.sender != processor {
+                return Err(ServiceError::Unauthorized(
+                    UnauthorizedReason::NotAllowed {},
+                ));
+            }
+            let config = CONFIG.load(deps.storage)?;
+            actions::process_action(deps, env, info, action_msgs, config)
+        }
+        ExecuteMsg::UpdateConfig { new_config } => {
+            cw_ownable::assert_owner(deps.as_ref().storage, &info.sender)?;
+            let config = new_config.validate(deps.as_ref())?;
+            CONFIG.save(deps.storage, &config)?;
```

Comment:
Here is an example of an issue, `new_config` here is not a `ServiceConfig`, its `OptionalServiceConfig`.
Tho looking at the utils, this is also not clear there as well, 

```rust
#[cw_ownable_execute]
#[cw_serde]
pub enum ExecuteMsg<T, U> {
    ProcessAction(T),
    UpdateConfig { new_config: U },
    UpdateProcessor { processor: String },
}
```
`U` should be `OptionalServiceConfig`

(note @stiiifff , adding comments here might help, changing `U` to `TOptionalConfig` would be better, no way we can enforce it?)

The main problem is, this will not be detected in any test that doesn't include the manager, because usually in tests you will just pass the type you expect to be passed (like you did).
But here we are using generics, so its somewhat hidden, but even if we forced a specific type on the base implementation, you won't get it because you have a custom implementation.
This might lead to some issues in the future, if the tests will miss those cases.

Reply:
I think once we have the manager we should have a tests for that one too to make sure it works as expected

Reply:
Fixed with the refactor.